pkg/operator/ceph/file/dependent.go

```diff
@@ -49,3 +70,43 @@ func CephFilesystemDependents(clusterdCtx *clusterd.Context, clusterInfo *client
 
 	return deps, nil
 }
+
+// return subvolume groups that have 1 or more subvolumes present in them
+func subvolumeGroupDependents(clusterdCtx *clusterd.Context, clusterInfo *client.ClusterInfo, filesystem *v1.CephFilesystem) (*dependents.DependentList, error) {
+	baseErr := "failed to get Ceph subvolume groups containing subvolumes"
+
+	deps := dependents.NewDependentList()
+
+	svgs, err := client.ListSubvolumeGroups(clusterdCtx, clusterInfo, filesystem.Name)
+	if err != nil {
+		return deps, errors.Wrap(err, baseErr)
+	}
+
+	errs := []error{}
+	for _, svg := range svgs {
+		svs, err := client.ListSubvolumesInGroup(clusterdCtx, clusterInfo, filesystem.Name, svg.Name)
+		if err != nil {
+			errs = append(errs, errors.Wrapf(err, "failed to list subvolumes in subvolume group %q", svg.Name))
+		}
+
+		if svg.Name == indexSubvolumeGroupName {
+			// TODO: there are two mutually exclusive design options here:
+			//   1. this check is future proof in the event that other metadata subvolumes are added
+			//      to '_index' like 'clone' exists today (not sure how likely this is)
+			//      (this is what is implemented currently)
+			//   2. we can explicitly only ignore the 'clone' subvolume but block if there are
```

Comment:
Seems like we don't need to check for this unexpected corner case

Reply:
Yes, it's a possible corner case, the user can manually create subvolumes or csi volumes in the _index subvolumegroup. having a hard-coded check for `clone` might not be sufficient as ceph fs might add some other subvolumes in the future for this subvolumegroup. @kotreshhr @ajarr why not cephfs blocks users from creating subvolumes in the group like `_index` `_legacy` as it's only meant for internal usage?